catalog_impls/src/table_based.rs

```diff
@@ -130,18 +130,17 @@ impl TableBasedManager {
         let mut manager = Self {
             catalog_table: Arc::new(catalog_table),
             catalogs: HashMap::new(),
-            engine_proxy,
             schema_id_generator: Arc::new(SchemaIdGenerator::default()),
+            table_infos: Vec::new(),
         };
 
         manager.init().await?;
 
         Ok(manager)
     }
 
-    #[cfg(test)]
-    pub fn get_engine_proxy(&self) -> TableEngineRef {
-        self.engine_proxy.clone()
+    pub fn get_table_infos(&self) -> Vec<TableInfo> {
+        self.table_infos.clone()
```

Comment:
I guess `table_infos` field is not a good idea because the field won't change with following drop/create table operation and that will lead to confusion.

Here, I'm in favor of using `VistorImpl` to fetch table infos dynamically from the backend engine.

Reply:
Ok, it's actually strange here.